File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    knight = []
    
    for i in range(1, number+1):
        if get_divisor_count(i) > limit:
            knight.append(power)
        else:
            knight.append(get_divisor_count(i))
    
    return sum(knight)
        
        
def get_divisor_count(n):
    end = int(n**0.5)
    count = 0
    for i in range(1, end+1):
        if n % i == 0:
            if i**2 == n:
                count += 1
            else:
                count += 2
    return count

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.swift
import Foundation

func solution(_ number:Int, _ limit:Int, _ power:Int) -> Int {
    return (1...number)
    .map {
        let attackPower = getDivisorCount($0)
        return attackPower > limit ? power : attackPower
    }.reduce(0, +)
}

func getDivisorCount(_ number: Int) -> Int {
    let end = Int(sqrt(Double(number)))
    var count = 0
    
    for i in 1...end {
        if number % i == 0 {
            count += i * i == number ? 1 : 2
        }
    }
    
    return count
}